### PR TITLE
feat: Custom renderer

### PR DESCRIPTION
**Pull request by adityathebe**
Addresses https://github.com/flanksource/incident-commander/issues/235

### CODE REVIEW

File: pkg/system_api.go [hunk 1]
```diff
@@ -166,6 +166,17 @@ func (components Components) Walk() Components {
 	return comps
 }
 
+type RenderComponent struct {
```
Comment (adityathebe) [inline pkg/system_api.go:169]: I created a new struct here. Wasn't sure if I should be using the existing `Component`

**Comment by moshloop** [comment]
@adityathebe this needs to go incident-commander, and the struct changes into duty, Nothing UI related should be in canary-checker, eventually all of the topology related code will move to incident commander, and only health checks will be performed by canary-checker.

File: pkg/topology/query.go [hunk 2]
```diff
@@ -198,6 +198,33 @@ func (p TopologyParams) GetComponentRelationWhereClause() string {
 	return s
 }
 
+func QueryRenderComponents(ctx context.Context, systemTemplateID string) ([]pkg.Renderers, error) {
+	query := `select spec->'renderers' from templates where id = $1`
+	args := map[string]interface{}{
+		"id": systemTemplateID,
+	}
+
+	rows, err := db.QueryNamed(ctx, query, args)
```
Comment (yashmehrotra) [inline pkg/topology/query.go:207]: Don't use `QueryNamed` function, prefer using Gorm or the underlying pool object to query

File: pkg/api/renderer.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+const componentRegistry = {
+	{{range $k, $v := .}}"{{$k}}": {{$k}},
+	{{end}}
+};
+`
+
+func init() {
+	tpl, err := template.New("registry").Parse(jsComponentRegistryTpl)
+	if err != nil {
+		panic(fmt.Errorf("error parsing template jsComponentRegistryTpl. %w", err))
```
Comment (yashmehrotra) [inline pkg/api/renderer.go:116]: Use logger.Fatalf instead of panic

File: pkg/api/renderer.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	if isProp {
+		prefix = "property"
+	}
+
+	if c.Type != "" {
+		return fmt.Sprintf("%s_%s_%s", prefix, c.Type, c.Name)
+	}
+
+	return fmt.Sprintf("%s_%s", prefix, c.Name)
+}
```
Comment (yashmehrotra) [inline pkg/api/renderer.go:91]: This can be a method of pkg.RenderComponent,  so we can use it like `renderComponent.Key(isProp)`

**Pull request closed without merging**